### ISSUE TITLE
Improve sharing: Change text and share geo link instead of om link.

### ISSUE THREAD

**Issue by gerion0** [organicmaps/organicmaps#2261]
When sharing an arbitratry location via Organic Maps with another app, the following text is displayed (german locale):

> Hi,

> ich bin gerade hier: Hauptbahnhof. Klicke den Link om://s4DZoF5cad/Hauptbahnhof oder https://omaps.app/s4DZoF5cad/Hauptbahnhof, um den Ort auf der Karte anzuzeigen.

> Vielen Dank.

Translation to english: "Hi, i'm currently here: ... Click on the link ... or ... to display the location on the map. Many thanks."
I have two comments to this format:
- I'm not necessarily on this location, when sharing it. Organic maps should be able to check this and change the text to "I want to share this location" or similar, when sharing an arbitrary location.
- The om:// link is not clickable. Would it be possible to replace it with a geo link? In this way, it is much more generic and should be of similar flexibility.

**Comment by vng** [comment]
https://github.com/organicmaps/organicmaps/pull/1821
@biodranik AFAIR, we share https://omaps.app/XXX links now right?

**Comment by biodranik** [comment]
It's not finished yet. But the issue above mentions incorrect translation for "my position" instead of another one for "any position".

**Comment by Zack-83** [comment]
As long as the sharing command generates Organic Maps links, users may share positions only with other users who already have the app installed. That restricts enormously the usability of the app!

In my humble opinion, it would be better to share more interoperable Open Street Maps links (either coordinate-based or object-based), together with a hint how to download Organic Maps and how to make it the default application for opening geographical links. That would help broadening the usage of open source maps software and also advertise Organic Maps in a friendly way.

**Comment by gerion0** [comment]
The geo links (geo:37.786971,-122.399677) that I mentioned are going in the same direction (they are even more generic). They are an [official URI scheme[(https://en.wikipedia.org/wiki/Geo_URI_scheme) and hopefully most maps apps should be able to open them (OM does that). I'm not sure, if the linked pull request also includes a geo link.

**Comment by patepelo** [comment]
Can we rescue this issue and focus on the priority of changing the wording? 

This is current:
>Hi,
I'm here now: _Lincoln Street, 3_ Click this link "om://" or this one "https://omaps.app/" to see the place on the map.

Proposal:
>McDonald's (Place Name if available) <br>
Lincoln Street, 3, 08037 Chicago (Full Address if available)<br>
48.390/20.978 (Coordinates in the most common and universal structure)
<br>geo:url (One link, preferably universal for any other navigation app)
Sent with Organic Maps. (Maybe adding a strap line)


WDYT?

**Comment by Misalf-git** [comment]
Yes, any wording that is **not part of the information** should be stripped.

IMO, several new-lines are not needed.
Also, when I use the share button, Androids preview of the shared content only displays the first line:
```
Hi, ...

```
and nothing else. Confusing.

I'd go with something like:

```
McDonald's: Lincoln Street, 3, 08037 Chicago
geo:48.390,20.978[...]
```

If it's really a necessity to include self-promotion (whith some pushy aftertaste) then `Sent with Organic Maps.` should be enough.

**Comment by pastk** [comment]
Please check the following PR, it was 80% done there, it needs a volunteer to pick it up and finish for the iOS
- https://github.com/organicmaps/organicmaps/pull/1821

**Comment by patepelo** [comment]
> Please check the following PR, it was 80% done there, it needs a volunteer to pick it up and finish for the iOS
> 
>     * [Location sharing text format update #1821](https://github.com/organicmaps/organicmaps/pull/1821)

So it's up to @organicmaps/ios to help out testing 🙏

**Comment by patepelo** [comment]
What's your opininion on having a universal geo: url?

**Comment by pastk** [comment]
AFAIR it was all discussed in that PR already and we've settled on the text and link formats there..

**Comment by Zack-83** [comment]
> Please check the following PR, it was 80% done there, it needs a volunteer to pick it up and finish for the iOS
> 
>     * [Location sharing text format update #1821](https://github.com/organicmaps/organicmaps/pull/1821)

I find that PR good, and it needs to be merged fastly, but it does not solve the present issue.
